File: algorithm/src/main/java/baekjoon/p01531/Main.java
package baekjoon.p01531;

import java.util.Scanner;

public class Main {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int m = sc.nextInt();
        int[][] inputs = new int[n][4];

        for (int i = 0; i < inputs.length; i++) {
            for (int j = 0; j < inputs[i].length; j++) {
                inputs[i][j] = sc.nextInt();
            }
        }

        System.out.println(solve(m, inputs));
    }

    private static int solve(int m, int[][] inputs) {
        int[][] arr = new int[101][101];
        for (int[] e : inputs) {
            inc(arr, e[0], e[1], e[2], e[3]);
        }

        int ret = 0;
        for (int[] e : arr) {
            for (int ee : e) {
                if (ee > m) {
                    ret++;
                }
            }
        }

        return ret;
    }

    private static void inc(int[][] arr, int y0, int x0, int y1, int x1) {
        for (int y = y0; y <= y1; y++) {
            for (int x = x0; x <= x1; x++) {
                arr[y][x]++;
            }
        }
    }

}

File: algorithm/src/main/java/baekjoon/p01531/Main.kt
package baekjoon.p01531

fun main(args: Array<String>) {
    val (n, m) = readLine()!!.split(" ").map { it.toInt() }
    val inputs = List(n) { readLine()!!.split(" ").map { it.toInt() } }
    println(solve(inputs, m))
}

private fun solve(inputs: List<List<Int>>, m: Int): Int {
    val arr = Array(101) { IntArray(101) }
    for (e in inputs) {
        inc(arr, e[0], e[1], e[2], e[3])
    }

    return arr.sumBy { it.count { it > m } }
}

private fun inc(arr: Array<IntArray>, y0: Int, x0: Int, y1: Int, x1: Int) {
    for (y in y0..y1) {
        for (x in x0..x1) {
            arr[y][x]++
        }
    }
}